src/Microsoft.Sbom.Api/Utils/PathPatternMatcher.cs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Sbom.Api.Utils;
+
+/// <summary>
+/// Provides pattern matching functionality for file paths using glob-style patterns.
+/// </summary>
+public static class PathPatternMatcher
```

Comment:
.NET already provides cross-platform support for globbing that we should leverage unless there's some reason that it doesn't do what we need it to do. See https://learn.microsoft.com/en-us/dotnet/core/extensions/file-globbing for details and syntax.